src/main/java/com/firebolt/jdbc/client/query/StatementClientImpl.java

```diff
@@ -290,6 +292,12 @@ protected void validateResponse(String host, Response response, Boolean isCompre
 		super.validateResponse(host, response, isCompress);
 		FireboltConnection connection = getConnection();
 		if (isCallSuccessful(response.code())) {
+			if (response.header(HEADER_UPDATE_ENDPOINT) != null) {
+				connection.setEndpoint(response.header(HEADER_UPDATE_ENDPOINT));
+			}
```

Comment:
Endpoint returned from the backend would contain query parameters e.g. `http://url.io/?account_id=1&database_name=b`. We want to parse it, do error checks (e.g. returned account id is the same that we hold) and I'd assume keep only the url part. Please verify with the design doc.

Reply:
I do not see this in spec:

PackDB appends the following HTTP headers to the response:
Firebolt-Update-Endpoint: https://new-endpoint.example.com/
Firebolt-Update-Parameters: engine=engine_name,account_id=account_id
Firebolt-Reset-Session:
client updates its map of parameters

Reply:
As discussed on the daily that's PackDB design doc, let's refer to the implementation doc that was created for Ecosystem specifically https://docs.google.com/document/d/1f0kdPCPbhCrmMTK-sfqAkqEkonT7xyNK-teJmXE_6p0/

Reply:
done